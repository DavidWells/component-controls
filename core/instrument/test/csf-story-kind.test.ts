import { loadStoriesTests } from './loadTestFiles';

describe('csf-story-kind', () => {
  loadStoriesTests('csf', 'kind');
});
