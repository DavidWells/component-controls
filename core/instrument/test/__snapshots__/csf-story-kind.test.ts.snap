// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`csf-story-kind no-title.js 1`] = `
Object {
  "components": Object {},
  "kinds": Object {
    "Story": Object {
      "components": Object {},
      "package": "2f97f282b46c08427abf79e3c3699074",
      "test": 1,
      "title": "Story",
    },
  },
  "packages": Object {
    "2f97f282b46c08427abf79e3c3699074": Object {
      "dependencies": Object {
        "@babel/generator": "^7.9.4",
        "@babel/parser": "^7.9.4",
        "@babel/traverse": "^7.9.0",
        "@component-controls/core": "^1.1.0",
        "@component-controls/specification": "^1.1.0",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "hosted-git-info": "^3.0.4",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "resolve": "^1.15.1",
        "typescript": "^3.8.3",
      },
      "devDependencies": Object {
        "@babel/types": "^7.9.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^25.1.2",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^24.9.0",
      },
      "fileHash": "2f97f282b46c08427abf79e3c3699074",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/csf/kind/no-title.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.1.0",
    },
  },
  "stories": Object {},
  "transformed": "export default {
  title: 'Story',
  test: 1,
};
",
}
`;

exports[`csf-story-kind title-and-parameters.js 1`] = `
Object {
  "components": Object {
    "21455c8d3b1413b6aaaa56c7ca6e1dde": Object {
      "name": "ControlsTable",
    },
  },
  "kinds": Object {
    "Storybook/Blocks/ControlsTable": Object {
      "component": "ControlsTable",
      "components": Object {
        "ControlsTable": "21455c8d3b1413b6aaaa56c7ca6e1dde",
      },
      "package": "e8764fc7a5e278ee187ee3d546896280",
      "parameters": Object {
        "addonControls": Object {
          "smart": false,
        },
        "component": "ControlsTable",
      },
      "title": "Storybook/Blocks/ControlsTable",
    },
  },
  "packages": Object {
    "e8764fc7a5e278ee187ee3d546896280": Object {
      "dependencies": Object {
        "@babel/generator": "^7.9.4",
        "@babel/parser": "^7.9.4",
        "@babel/traverse": "^7.9.0",
        "@component-controls/core": "^1.1.0",
        "@component-controls/specification": "^1.1.0",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "hosted-git-info": "^3.0.4",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "resolve": "^1.15.1",
        "typescript": "^3.8.3",
      },
      "devDependencies": Object {
        "@babel/types": "^7.9.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^25.1.2",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^24.9.0",
      },
      "fileHash": "e8764fc7a5e278ee187ee3d546896280",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/csf/kind/title-and-parameters.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.1.0",
    },
  },
  "stories": Object {},
  "transformed": "export default {
  title: 'Storybook/Blocks/ControlsTable',
  parameters: {
    component: ControlsTable,
    addonControls: {
      smart: false,
    },
  },
};
",
}
`;

exports[`csf-story-kind title-controls-and-parameters.js 1`] = `
Object {
  "components": Object {
    "dec877e66c549940695c423429a5d00b": Object {
      "name": "ControlsTable",
    },
  },
  "kinds": Object {
    "Storybook/Kind": Object {
      "component": "ControlsTable",
      "components": Object {
        "ControlsTable": "dec877e66c549940695c423429a5d00b",
      },
      "package": "2edbb8908aafc7a0c993dc189a899134",
      "parameters": Object {
        "controls": Object {
          "name": Object {
            "label": "Name",
            "order": 9999,
            "type": "ControlTypes.TEXT",
            "value": "Mark",
          },
        },
      },
      "title": "Storybook/Kind",
    },
  },
  "packages": Object {
    "2edbb8908aafc7a0c993dc189a899134": Object {
      "dependencies": Object {
        "@babel/generator": "^7.9.4",
        "@babel/parser": "^7.9.4",
        "@babel/traverse": "^7.9.0",
        "@component-controls/core": "^1.1.0",
        "@component-controls/specification": "^1.1.0",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "hosted-git-info": "^3.0.4",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "resolve": "^1.15.1",
        "typescript": "^3.8.3",
      },
      "devDependencies": Object {
        "@babel/types": "^7.9.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^25.1.2",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^24.9.0",
      },
      "fileHash": "2edbb8908aafc7a0c993dc189a899134",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/csf/kind/title-controls-and-parameters.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.1.0",
    },
  },
  "stories": Object {},
  "transformed": "export default {
  title: 'Storybook/Kind',
  component: ControlsTable,
  parameters: {
    controls: {
      name: {
        type: ControlTypes.TEXT,
        label: 'Name',
        value: 'Mark',
        order: 9999,
      },
    },
  },
};
",
}
`;

exports[`csf-story-kind with-title.js 1`] = `
Object {
  "components": Object {},
  "kinds": Object {
    "Storybook/Blocks/ControlsTable": Object {
      "components": Object {},
      "package": "a381fc9b47b2f128ff40a172faed3d04",
      "title": "Storybook/Blocks/ControlsTable",
    },
  },
  "packages": Object {
    "a381fc9b47b2f128ff40a172faed3d04": Object {
      "dependencies": Object {
        "@babel/generator": "^7.9.4",
        "@babel/parser": "^7.9.4",
        "@babel/traverse": "^7.9.0",
        "@component-controls/core": "^1.1.0",
        "@component-controls/specification": "^1.1.0",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "hosted-git-info": "^3.0.4",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "resolve": "^1.15.1",
        "typescript": "^3.8.3",
      },
      "devDependencies": Object {
        "@babel/types": "^7.9.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^25.1.2",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^24.9.0",
      },
      "fileHash": "a381fc9b47b2f128ff40a172faed3d04",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/csf/kind/with-title.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.1.0",
    },
  },
  "stories": Object {},
  "transformed": "export default {
  title: 'Storybook/Blocks/ControlsTable',
};
",
}
`;
